CorsixTH/Lua/rooms/ward.lua

```diff
@@ -135,14 +153,20 @@ function WardRoom:commandEnteringPatient(patient)
     print("Warning: A patient was called into the ward even though there are no free beds.")
   else
     bed.reserved_for = patient
-    self:countWorkingNurses()
-    local length = (math.random(200, 800) * (1.5 - staff.profile.skill))  / self.nursecount -- reduce time in ward if there is more than one nurse on duty
+    local amount_to_heal = math.random(150, 450)
     local --[[persistable:ward_loop_callback]] function loop_callback(action)
-    -- TODO Perhaps it should take longer if there are more used beds!
-      if length <= 0 then
+      if not self.patient_idx then -- This is the first patient for the current tick
+        self:getHealingAmount() -- Evaluate the workload
+      end
+      amount_to_heal = amount_to_heal - self.healing_amount
+      if amount_to_heal < 0 then -- The patient is healed or diagnosed
         action.prolonged_usage = false
       end
-      length = length - 1
+      if self.patient_idx == 1 then -- This was the last patient, set up for the next tick
+        self.patient_idx = nil -- Erase the temporary field
+      else -- Set up for the next patient
+        self.patient_idx = self.patient_idx - 1
+      end
     end
     local after_use = --[[persistable:ward_after_use]] function()
```

Comment:
Nothing wrong inserting an empty line when you start doing a next step. Such lines guide the reader on such context switches in the same way that you add an empty line between different pieces of text in a message.